bindings/node.js/kzg.cxx

```diff
@@ -0,0 +1,366 @@
+  Napi::TypeError::New(
+    env,
+    "Invalid parameter type: " + name + ". Expected " + expectedType
+  ).ThrowAsJavaScriptException();
+
+  return env.Null();
+}
+
+Napi::TypedArrayOf<uint8_t> napi_typed_array_from_bytes(uint8_t* array, size_t length, Napi::Env env) {
+  // Create std::vector<uint8_t> out of array.
+  // We allocate it on the heap to allow wrapping it up into ArrayBuffer.
+  std::unique_ptr<std::vector<uint8_t>> vector =
+      std::make_unique<std::vector<uint8_t>>(length, 0);
+
+  for (size_t i = 0; i < length; ++i) {
+    (*vector)[i] = array[i];
+  }
+
+  // Wrap up the std::vector into the ArrayBuffer.
+  Napi::ArrayBuffer buffer = Napi::ArrayBuffer::New(
+      env,
+      vector->data(),
+      length /* size in bytes */,
+      [](Napi::Env /*env*/, void* /*data*/, std::vector<uint8_t>* hint) {
+        std::unique_ptr<std::vector<uint8_t>> vectorPtrToDelete(hint);
+      },
+      vector.get());
+
+  // The finalizer is responsible for deleting the vector: release the
+  // unique_ptr ownership.
+  vector.release();
+
+  return Napi::Uint8Array::New(env, length, buffer, 0);
+}
+
+// loadTrustedSetup: (filePath: string) => SetupHandle;
+Napi::Value LoadTrustedSetup(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 1;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  if (!info[0].IsString()) {
+    return throw_invalid_argument_type(env, "filePath", "string");
+  }
+
+  const std::string file_path = info[0].ToString().Utf8Value();
+
+  KZGSettings* kzg_settings = (KZGSettings*)malloc(sizeof(KZGSettings));
+
+  if (kzg_settings == NULL) {
+    Napi::Error::New(env, "Error while allocating memory for KZG settings").ThrowAsJavaScriptException();
+    return env.Null();
+  };
+
+  FILE* f = fopen(file_path.c_str(), "r");
+
+  if (f == NULL) {
+    free(kzg_settings);
+    Napi::Error::New(env, "Error opening trusted setup file").ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  if (load_trusted_setup(kzg_settings, f) != C_KZG_OK) {
+    free(kzg_settings);
+    Napi::Error::New(env, "Error loading trusted setup file").ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  return Napi::External<KZGSettings>::New(info.Env(), kzg_settings);
+}
+
+// freeTrustedSetup: (setupHandle: SetupHandle) => void;
+Napi::Value FreeTrustedSetup(const Napi::CallbackInfo& info) {
+  auto env = info.Env();
+
+  size_t argument_count = info.Length();
+  size_t expected_argument_count = 1;
+  if (argument_count != expected_argument_count) {
+    return throw_invalid_arguments_count(expected_argument_count, argument_count, env);
+  }
+
+  auto kzg_settings = info[0].As<Napi::External<KZGSettings>>().Data();
+  free_trusted_setup(kzg_settings);
+  free(kzg_settings);
```

Comment:
I thought `free_trusted_setup` already does this `free`?

Reply:
`free_trusted_setup` frees only the fields within `kzg_settings`.